earthmover/earthmover.py

```diff
@@ -434,6 +448,15 @@ def generate(self, selector):
                 node.num_rows = num_rows
 
             active_graph.draw()
+        
+        ### Create structured output results_file if necessary
+        if self.results_file:
+            self.end_timestamp = datetime.datetime.now()
+            self.metadata.update({"completed_at": self.end_timestamp.isoformat(timespec='microseconds')})
+            self.metadata.update({"runtime_sec": (self.end_timestamp - self.start_timestamp).total_seconds()})
+            with open(self.results_file, 'w') as fp:
+                fp.write(json.dumps(self.metadata, indent=4))
+                fp.close()

```

Comment:
The `with` keyword here opens a context manager that closes automatically. We don't need line 459.